R/as.IAMCTimeseries.R

```diff
@@ -0,0 +1,49 @@
+#' Converts an IAMC long format into a wide format
+#' 
+#' Pivots periods to columns so that values are given as a time series table rather than a long list.
+#'
+#' @md
+#' @param df A [`quitte`] data frame
+#' @param fill A fill value. Defaults to `NA`
+#'
+#' @author Michaja Pehl, Tonn Rüter
+#'
+#' @examples
+#' \dontrun{
+#'   qf <- read.quitte("path/to/file") %>% as.IAMCTimeseries()
+#' }
+#'
+#' @importFrom rlang .data
+#' @importFrom dplyr rename_with select
+#' @importFrom tidyr pivot_wider
+#' @importFrom tidyselect matches
+#' @importFrom stringr str_to_title
+#' @export
+as.IAMCTimeseries <- function(df, fill = NA) {
+  # Silence "no visible binding for global variable '.'" build problems. We'll use colnames(.) later on. When using
+  # . (dot) in a magrittr pipe, it refers to whatever is on the left hand side of %>%
+  . <- NULL
+  default_columns <- c("Model", "Scenario", "Region", "Variable", "Unit", "Period", "Value")
+  return(
+    as.quitte(df) %>%
```

Comment:
Also, this runs afoul of the S3 object "system" (ironic air quotes) we discussed on Wednesday.
`as.quitte()` (besides being too restrictive on the input data for my personal taste) makes the data of class `quitte`, which it most assuredly is not anymore after pivoting the periods -- but the class attribute still says it is.
So either extend the class definition and all funtions relying on it, or slap an `unquitte()` after the pivot. ;)